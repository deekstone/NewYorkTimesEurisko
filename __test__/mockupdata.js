
export const rowItem = {
    "abstract":"From Hong Kong to Kuwait to Paris, crowds gathered to watch U.S. presidential election results and supporters of Barack Obama celebrated his victory.",
    "web_url":"https://www.nytimes.com/slideshow/2008/11/05/world/1105-REACTS_index.html",
    "snippet":"From Hong Kong to Kuwait to Paris, crowds gathered to watch U.S. presidential election results and supporters of Barack Obama celebrated his victory.",
    "lead_paragraph":"From Hong Kong to Kuwait to Paris, crowds gathered to watch U.S. presidential election results and supporters of Barack Obama celebrated his victory.",
    "source":"The New York Times",
    "multimedia":[
       {
          "rank":0,
          "subtype":"wide",
          "caption":null,
          "credit":null,
          "type":"image",
          "url":"images/2008/11/05/world/1105-REACTS-B.JPG",
          "height":126,
          "width":190,
          "legacy":{
             "widewidth":190,
             "wideheight":126,
             "wide":"images/2008/11/05/world/1105-REACTS-B.JPG"
          },
          "subType":"wide",
          "crop_name":"thumbWide"
       },
       {
          "rank":0,
          "subtype":"thumbnail",
          "caption":null,
          "credit":null,
          "type":"image",
          "url":"images/2008/11/05/world/1105-REACTS-S.JPG",
          "height":75,
          "width":75,
          "legacy":{
             "thumbnail":"images/2008/11/05/world/1105-REACTS-S.JPG",
             "thumbnailwidth":75,
             "thumbnailheight":75
          },
          "subType":"thumbnail",
          "crop_name":"thumbStandard"
       }
    ],
    "headline":{
       "main":"The World Reacts",
       "kicker":null,
       "content_kicker":null,
       "print_headline":null,
       "name":null,
       "seo":null,
       "sub":null
    },
    "keywords":[
       {
          "name":"subject",
          "value":"Presidential Election of 2008",
          "rank":1,
          "major":"N"
       },
       {
          "name":"persons",
          "value":"Obama, Barack",
          "rank":2,
          "major":"N"
       },
       {
          "name":"subject",
          "value":"International Relations",
          "rank":3,
          "major":"N"
       }
    ],
    "pub_date":"2011-04-20T04:02:07+0000",
    "document_type":"multimedia",
    "news_desk":"World",
    "section_name":"World",
    "byline":{
       "original":null,
       "person":[
          
       ],
       "organization":null
    },
    "type_of_material":"Slideshow",
    "_id":"nyt://slideshow/ec223483-91c7-5f5e-bf7b-f90f1c441aaa",
    "word_count":0,
    "slideshow_credits":"Ruth Fremson/The New York Times",
    "uri":"nyt://slideshow/ec223483-91c7-5f5e-bf7b-f90f1c441aaa"
 }