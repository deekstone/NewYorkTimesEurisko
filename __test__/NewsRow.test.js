// __tests__/Intro-test.js
import React from 'react';
import NewsRow from '../src/components/news/NewsRow';
import NewsDetail from '../src/components/newsDetail/NewsDetail';
import renderer from 'react-test-renderer';
import {rowItem} from './mockupdata'
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('renders correctly', () => {
  const tree = renderer.create(<NewsRow rowItem={rowItem} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Image element must appear in case the news has an image', () => {
  const wrapper = shallow(<NewsRow rowItem={rowItem} />);
  expect(wrapper.find('Image').length).toEqual(1);
});



