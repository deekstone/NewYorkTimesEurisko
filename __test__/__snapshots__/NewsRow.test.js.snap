// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "elevation": 5,
      "marginBottom": 10,
      "overflow": "hidden",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <View
        collapsable={false}
      >
        <Image
          source={
            Object {
              "uri": "https://www.nytimes.com/images/2008/11/05/world/1105-REACTS-B.JPG",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 180,
              "resizeMode": "cover",
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "backgroundColor": "rgba(52, 52, 52, 0.8)",
            "bottom": 0,
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
            "padding": 10,
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        The World Reacts
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#545454",
          "fontSize": 16,
          "lineHeight": 22,
          "padding": 10,
          "paddingTop": 8,
        }
      }
    >
      From Hong Kong to Kuwait to Paris, crowds gathered to watch U.S. presidential election results and supporters of Barack Obama celebrated his victory.
    </Text>
  </View>
</View>
`;
